cpp/ql/src/Security/CWE/CWE-311/CleartextFileWrite.ql

```diff
@@ -15,10 +15,37 @@ import cpp
 import semmle.code.cpp.security.SensitiveExprs
 import semmle.code.cpp.security.FileWrite
 import semmle.code.cpp.dataflow.DataFlow
+import semmle.code.cpp.valuenumbering.GlobalValueNumbering
+
+/**
+ * An operation on a filename.
+ */
+predicate filenameOperation(FunctionCall op, Expr path) {
```

Comment:
Could this be shared with the TOCTOU query somehow, or are they too different?

Reply:
Yes, now that the TOCTOU changes are merged I should be able to do this...

Reply:
On reflection I think the right way to do this is to add a new 'FileOperation' or 'Files' model to 'models', with methods such as `getAFilenameParameter` (and perhaps `getAFileDescriptorParameter`) exposing the information we need.  I'd rather do this in a separate PR, preferably after we've had the public/private model implementations discussion as well.

Alternatively I could just move the `filenameOperation`, `accessCheck` and `stat` predicates into a common library, but I'm not sure where is appropriate, and we'll end up maintaining this interface.

Reply:
Ticket for this: https://github.com/github/codeql-c-team/issues/607

Reply:
Let's wait for that discussion to finish and do it in a separate PR.